src/components/dashboard/SendLinkSummary.js

```diff
@@ -69,43 +71,20 @@ const SendLinkSummary = ({ screenProps, styles }: AmountProps) => {
             message: `You can still copy the link by tapping on "Copy link to clipboard".`,
             dismissText: 'Ok',
             onDismiss: () => {
-              const desktopShareLink = generateSendShareText(
-                paymentLink,
-                amount,
-                counterPartyDisplayName,
-                profile.fullName
-              )
-
-              screenProps.push('SendConfirmation', {
+              const desktopShareLink = generateSendShareText(...shareStringSource)
```

Comment:
why spread is needed here ? is generateSendShareText mutates params ? if yes - this should be fixed 
Becuase generateSendShareText should just take some params and return string, nothing more
in other words it should be a **pure function**

https://reactjs.org/docs/components-and-props.html#props-are-read-only
https://en.wikipedia.org/wiki/Pure_function

Reply:
@serdiukov-o-nordwhale no it's not mutating value. It just used the values to build share string. I am using spread to pass all of the array items as separate arguments just as it was before.

Reply:
aah, it just function call with array of arguments. ok, sorry